BinaryObjectScanner/Protection/SmartE.cs

```diff
@@ -1,28 +1,47 @@
 ﻿using System.Collections.Generic;
 using BinaryObjectScanner.Interfaces;
 using SabreTools.Matching;
+using SabreTools.Matching.Content;
 using SabreTools.Matching.Paths;
 using SabreTools.Serialization.Wrappers;
 
 namespace BinaryObjectScanner.Protection
 {
-    public class SmartE : IExecutableCheck<PortableExecutable>, IPathCheck
+    public class SmartE : IPathCheck, IExecutableCheck<PortableExecutable>
     {
-        /// <inheritdoc/>
         public string? CheckExecutable(string file, PortableExecutable pex, bool includeDebug)
         {
-            // Get the last section strings, if they exist
-            var sections = pex.Model.SectionTable ?? [];
-            var strs = pex.GetSectionStrings(sections.Length - 1);
-            if (strs != null)
-            {
-                if (strs.Exists(s => s.Contains("BITARTS")))
-                    return "SmartE";
-            }
+           // Only works on stub generated from running the program yourself
+           if (pex.InternalName.OptionalEquals("SmarteSECURE"))
+               return "SmartE";
+           
+           var sections = pex.Model.SectionTable ?? [];
 
-            return null;
+           if (sections.Length > 0)
+           {
+               // Get the last section data, if it exists
+               var lastSectionData = pex.GetSectionData(sections.Length - 1);
+               if (lastSectionData != null)
+               {
+                   // All sections seen so far are the last sections, so this is "technically"
+                   // the only known needed check so far. Others kept as backups if this fails
+                   // on some future entry
+                   var matchers = GenerateMatchers();
+                   var match = MatchUtil.GetFirstMatch(file, lastSectionData, matchers, includeDebug);
+                   if (!string.IsNullOrEmpty(match))
+                       return match;
+               }
+           }
+           
+           // Specific known named sections:
+           // .bss (Rise of Nations)
+           // .tls (Zoo Tycoon 2)
+           // .idata (http://redump.org/disc/58561/ and http://redump.org/disc/71983/)
+           // .edata (http://redump.org/disc/36619/)
+           
+           return null;
         }
-
+        
```

Comment:
Last nit: get rid of the unnecessary whitespace on this line

Reply:
Fixed.